label_studio_ml/examples/yolo/README_TIMELINE_LABELS.md

```diff
@@ -0,0 +1,517 @@
+  ```
+
+### Trainable Mode
+
+The trainable mode enables the model to learn from your annotations incrementally.
+It uses the pretrained YOLO classification model and a custom LSTM neural network on the top 
+to capture temporal dependencies in video data. The LSTM model works from scratch, 
+so it requires about 10-20 well-annotated videos 500 frames each (~20 seconds) to start making meaningful predictions.
+
+- **When to Use**: When custom labels or improved accuracy are needed relatively to simple mode.
+- **Configuration**: Set `model_trainable="true"` in the labeling config.
+- **Training Process**:
+  - Start annotating videos using the TimelineLabels tag.
+  - After submitting the first annotation, the model begins training.
+  - The `partial_fit()` method allows the model to train incrementally with each new annotation.
+- **Requirements**: Approximately 10-20 annotated tasks are needed to achieve reasonable performance.
+
+**Note**: The `predicted_values` attribute in the `<Label>` tag doesn't make sense for trainable models.
+
+**Example**:
+
+```xml
+<View>
+    <Video name="video" value="$video" height="700" timelineHeight="200" frameRate="25.0" />
+    <TimelineLabels name="label" toName="video"
+                    model_trainable="true"
+                    model_classifier_epochs="1000"
+                    model_classifier_sequence_size="16"
+                    model_classifier_hidden_size="32"
+                    model_classifier_num_layers="1"
+                    model_classifier_f1_threshold="0.95"
+                    model_classifier_accuracy_threshold="0.99"
+                    model_score_threshold="0.5">
+      <Label value="Ball in frame"/>
+      <Label value="Ball touch"/>
+    </TimelineLabels>
+</View>
+```
+
+## How the trainable model works
+
+The trainable mode uses a custom implementation of the temporal LSTM classification model. 
+The model is trained incrementally with each new annotation submit or update, 
+and it generates predictions for each frame in the video.
+
+### 1. Feature Extraction with YOLO
+
+- **Pre-trained YOLO Model**: Uses a YOLO classification model (e.g., `yolov8n-cls.pt`) to extract features from video frames.
+- **Layer Modification**: The model removes the last classification layer to use the feature representations from the penultimate layer (see `utils/neural_nets.py::cached_feature_extraction()`).
+- **Cached Predictions**: Uses caching to store YOLO intermediate feature extractions for efficiency and incremental training on the fly.
+
+#### Custom YOLO models for feature extraction
+
+You can load your own YOLO models using the steps described in the [main README](./README.md#custom-yolo-models).
+However, it should have similar architecture as `yolov8-cls` models. See `utils/neural_nets.py::cached_feature_extraction()` for more details.
+
+#### Cache folder 
+
+It's located in `/app/cache_dir` and stores the cached intermediate features from the last layer of the YOLO model. 
```

Comment:
I think this folder only exists inside the docker container. Do we want to leave it here, or remove?